README.adoc

```diff
@@ -1,5 +1,228 @@
-judo-generator-commons
-----------------------
+# Judo Generator Commons
 
 JUDO Generator Commons project contains helpers, annotations and maven utils which can be used on a generator engine development
 
+This is a general representation of the generator, which will be parameterized using one of the meta-models.
+We use the generator in these meta models:
+
+- judo-meta-esm
+- judo-meta-pam
+- judo-meta-ui
+
+There are 3 modules in these projects, which are related to the generator commons:
+
+- generator-engine
+- generator-maven-project
+- generator-maven-plugin-test
+
+The `generator-engine` teaches the generator commons how to release the items. This bind the meta model to the generator.
+Models are transformed into template parameters in this project.
+
+
+For example in the judo-meta-esm :
+
+    EsmGenerator.java -> Process of parameterization.
+
+    EsmGeneratorParameter.java -> This contains the model as a parameter, the generatorContext, the DirectoryResolvers etc.
+
+=== Description of a project
+
+The <project type>.yaml file describes what template is used for that and control which parameters are passed and the model can control HOW the templates can be used.
+Each generator-engine project has test directory with test project. This test project provides an example of how a `test-project.yaml` description looks like.
+
+Typically struct is :
+
+[source,yaml]
+----
+templates:
+
+  - name: testOverride
+    pathExpression: "#actorType.name + '/actorToOverride'"
+    templateName: test1/actorToOverride.hbs
+    actorTypeBased: true
+
+  - name: testReplace
+    pathExpression: "#actorType.name + '/actorToReplace'"
+    templateName: test1/actorToReplace.hbs
+    actorTypeBased: true
+
+  - name: testDelete
+    pathExpression: "#actorType.name + '/actorToDelete'"
+    templateName: test1/actorToDelete.hbs
+    actorTypeBased: true
+----
+
+* `templates`: This defines a list of templates.
+
+* `name`: This is like an id for the template. Based on this, we can override the template later.
+
+* `pathExpression`: This specifies where the generated file is located and under what name it will be generated. We use spring expression language (https://docs.spring.io/spring-framework/docs/3.0.x/reference/expressions.html[springEL]), because handlebars only support string output.
+
```

Comment:
The SpringEL is not only for pathExpression. Have to mention in an expression block. Because factoryExpression, condtionExpresison and templateContext  also uses. So in yaml every expression is SpringEL

Reply:
[624742d](https://github.com/BlackBeltTechnology/judo-generator-commons/pull/29/commits/624742d19b43c9565616a062820dd6a187057c29)